smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/KafkaSource.java

```diff
@@ -81,7 +83,10 @@ public KafkaSource(Vertx vertx,
         }
 
         configuration = config;
-        context = vertx.getOrCreateContext();
+        // We cannot use vertx.getOrCreate context as it would retrieve the same one everytime.
+        // It associates the context with the caller thread which will always be the same.
+        // So, we force the creation of different event loop context.
```

Comment:
Event loop contexts are load-balanced between event loop threads, IIUC?

Reply:
Or actually -- each event loop context is assigned to an event loop thread (in a round robin way I guess?) and stays that way. I think?

Reply:
Yes, exactly. 

An event loop context is associated with a specific event loop thread and will stay on that event loop thread. 
If you create more contexts than the number of event loops threads (generally 2 * the number of CPU Core), you will have multiple contexts associated with the same event loop thread.

Reply:
Cool, didn't know that, thanks!